src/Renci.SshNet/SftpClient.cs

```diff
@@ -801,10 +801,15 @@ public void UploadFile(Stream input, string path, bool canOverride, Action<ulong
 
             var flags = Flags.Write | Flags.Truncate;
 
-            if (canOverride)
-                flags |= Flags.CreateNewOrOpen;
+            if (input.Position > 0)
+                flags = Flags.Write | Flags.Append;
```

Comment:
File upload: If the local stream position is not zero, then open the remote file in APPEND mode